### PR TITLE
[CBRD-23214] Refactor client type definition

### PR DESCRIPTION
**Pull request by razvanradulescu**
http://jira.cubrid.org/browse/CBRD-23214

Add db_client_type.hpp.
Replace usage of BOOT_CLIENT_.. with DB_CLIENT_TYPE_..
Use enum values in cas_execute.c

### CODE REVIEW

File: src/transaction/client_credentials.hpp [hunk 1]
```diff
@@ -25,31 +25,14 @@
...
-  BOOT_CLIENT_ADMIN_CSQL = 8,
-  BOOT_CLIENT_LOG_COPIER = 9,
-  BOOT_CLIENT_LOG_APPLIER = 10,
-  BOOT_CLIENT_RW_BROKER_REPLICA_ONLY = 11,
-  BOOT_CLIENT_RO_BROKER_REPLICA_ONLY = 12,
-  BOOT_CLIENT_SO_BROKER_REPLICA_ONLY = 13,
-  BOOT_CLIENT_ADMIN_CSQL_WOS = 14,	/* admin csql that can write on standby */
-};
-typedef enum boot_client_type BOOT_CLIENT_TYPE;
+typedef enum db_client_type boot_client_type;
```
Comment (razvanradulescu) [inline src/transaction/client_credentials.hpp]: I tried :

> using  boot_client_type = db_client_type;
or 
> using  boot_client_type = enum db_client_type;

But the VS compiler complains about assigning enum of db_client_type to value of boot_client_type.

File: src/compat/db_admin.c [hunk 2]
```diff
@@ -112,6 +112,8 @@ static DB_HOST_STATUS_LIST db_Host_status_list;
 static DB_HOST_STATUS *db_add_host_status (char *hostname, int status);
 static DB_HOST_STATUS *db_find_host_status (char *hostname);
 
+int db_Client_type = DB_CLIENT_TYPE_DEFAULT;
```
Comment (ribram) [inline src/compat/db_admin.c]: Is this static? If it is, please also add static keyword to make the storage scope clear.

File: src/transaction/client_credentials.hpp (hunk 1 above)
Comment (ribram) [reply src/transaction/client_credentials.hpp]: Do you need both aliases boot_client_type and BOOT_CLIENT_TYPE? I think lowercase alias may be replaced with db_client_type directly.

Comment (razvanradulescu) [reply src/transaction/client_credentials.hpp]: both boot_client_type and BOOT_CLIENT_TYPE
are used in the code.
I preferred to not change the existing code to keep only one

**Pull request merged**